### PR TITLE
deploy.py script to replace run-playbook.sh

### LINKED ISSUE #261: Rewrite `run-playbook.sh` in Python
We could have this script:
- Check whether Bitwarden is unlocked before starting a run to prevent immediately crashing (#250)
- Check and validate required parameters
- Don't be Bash
- Have a flag to prevent sending chat spam and/or allow sending chat spam to the deploying user instead of `#notifications`

### PR DESCRIPTION
**Pull request by Riscky**
This closes #261 

This is a complete, nearly drop-in replacement of run-playbook.sh
This should pave the way for using a more modular playbook (probably with roles), see #315

The most important differences are:
- Nice Click interface
- Sends Discord messages without slacktee
- Stronger checks on worktree and branch consistency
- Doesn't leak environment variables

Optional TODO's include:
- [x] Running Black on the file
- [x] Typechecking (MyPy)
- [x] Putting in a shebang for less typing
- [x] `--force` flag to override checks
- [ ] Add black and mypy to Nixfile

Feedback is welcome :)

### CODE REVIEW

File: ansible/deploy.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
+import click
+import git
+import os
+import subprocess
+import requests
+import json
+
+
+@click.command()
+@click.option('--host', required=True, type=click.Choice(['staging', 'production']), help='The host to deploy to')
```
Comment (maartenberg) [inline ansible/deploy.py:10]: Suggestion: if you add an alias `--to` you can then do `python deploy.py --to production`, which might be a little nicer / easier to remember than `python deploy.py --host production`.

```suggestion
@click.option('--host', '--to', required=True, type=click.Choice(['staging', 'production']), help='The host to deploy to')
```

File: ansible/deploy.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
+import click
```
Comment (maartenberg) [inline ansible/deploy.py:1]: I suggest making this file executable and adding a shebang so you can invoke it as `./deploy.py`:

```suggestion
#!/usr/bin/env python
import click
```

File: ansible/deploy.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+  if r.status_code != 200:
+    raise click.ClickException(f"Discord failed with statuscode {r.status_code} and message {r.text}")
+
+def verify_on_latest_master(host):
+  repo = git.Repo('..')
+
+  # not sure why str() needs to be called
+  if host == 'production' and str(repo.active_branch) != 'master':
+    raise click.ClickException('Please do not deploy anything but master to production')
```
Comment (maartenberg) [inline ansible/deploy.py:136]: We probably want a way to disable this check if we ever need to deploy a non-master branch in case of trouble.

**Review by maartenberg — COMMENTED** [review]
Some initial comments.
Additional general notes:

 - Please run `black` over the file so we have a consistent code style for Python code, it's already in the Nix environment. We might also want to consider running it automatically in GH Actions.
 - Please add type signature information to the functions in this file, this makes it possible to use `mypy` for type checking.

File: ansible/deploy.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+import git
+import os
+import subprocess
+import requests
+import json
+
+
+@click.command()
+@click.option('--host', required=True, type=click.Choice(['staging', 'production']), help='The host to deploy to')
+@click.option('--playbook', default='main.yml', help='Ansible playbook to run')
```
[the remark this answers is not in the document]
Comment (Riscky) [reply ansible/deploy.py]: nice

File: ansible/deploy.py (hunk 1 above)
Comment (Riscky) [reply ansible/deploy.py]: I love this one!

**Review by maartenberg — APPROVED** [review]
:shipit:

**Pull request merged**